### PR TITLE
Remove deprecated `checkpoint_callback` flag in Trainer

### PR DESCRIPTION
**Pull request by shenoynikhil**
Part of #12521 

## What does this PR do?
This PR removes the `checkpoint_callback` flag from the Trainer Class. Following edits happen,
- pytorch_lightning/trainer/trainer.py
- pytorch_lightning/trainer/connectors/callback_connector.py
- CHANGELOG
- removed test from tests/deprecated_api/test_remove_1-7.py
- updated line from [simple_classif_training.py](https://github.com/PyTorchLightning/pytorch-lightning/pull/13027/commits/19922113f39a3229eebc3209539e5aedcf86daf6)

### Does your PR introduce any breaking changes? If yes, please list them.
```diff
- Trainer(checkpoint_callback=False)
+ Trainer(enable_checkpointing=False)
```

resolves #9754

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by shenoynikhil** [comment]
#### Dependence in Other Important Repostories
Based on @kaushikb11, I did a search on grep.app and found multiple important examples from 
- huggingface [[Example 1](https://github.com/huggingface/transformers/blob/6d80c92c77593dc674052b5a46431902e6adfe88/examples/research_projects/seq2seq-distillation/finetune.py#L414)] [[Example 2](https://github.com/huggingface/transformers/blob/6d80c92c77593dc674052b5a46431902e6adfe88/examples/research_projects/seq2seq-distillation/lightning_base.py#L384)]
- NVIDIA [[Example 1](https://github.com/NVIDIA/DeepLearningExamples/blob/8284bf4946db7da25ed3213f2e75e24828877e5b/PyTorch/LanguageModeling/BART/lightning_base.py#L372)]
Is this something we still plan to do?

**Comment by shenoynikhil** [comment]
@akihironitta This is my first contribution to PL, let me know if I have missed something obvious :).

**Review by akihironitta — APPROVED** [review]
Thanks for working on this! LGTM!

File: legacy/simple_classif_training.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def main_train(dir_path, max_epochs: int = 20):
         default_root_dir=dir_path,
         gpus=int(torch.cuda.is_available()),
         precision=(16 if torch.cuda.is_available() else 32),
-        checkpoint_callback=True,
+        enable_checkpointing=True,
```
[the remark this answers is not in the document]
Comment (shenoynikhil98) [reply legacy/simple_classif_training.py]: Sure! Added a commit with this change

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
